driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientEdgeIterator.java

```diff
@@ -88,7 +88,7 @@ public OrientEdge createGraphElement(final Object iObject) {
                 edge = new OrientEdge(this.sourceVertex.getGraph(), rec.getIdentity(), (OIdentifiable) this.sourceVertex.id(),
                     connection.getValue());
         } else if (immutableSchema.isEdgeType()) {
-            edge = new OrientEdge(this.sourceVertex.getGraph(), rec.getIdentity(), connection.getValue());
+            edge = new OrientEdge(this.sourceVertex.getGraph(), rec.getRecord(), connection.getValue());
```

Comment:
was this change really necessary? loading the record will take longer than just the ID - the idea is to only load the record if necessary.